packages/nextlove/src/types/index.ts

```diff
@@ -98,9 +104,30 @@ export type RouteFunction<
             : {}
         }
     : `unknown auth type: ${RS["auth"]}. You should configure this auth type in your auth_middlewares w/ createWithRouteSpec, or maybe you need to add "as const" to your route spec definition.`,
-  res: NextApiResponse<
-    RS["jsonResponse"] extends z.ZodTypeAny ? z.infer<RS["jsonResponse"]> : any
-  >
+  res: NextApiResponseWithoutJsonAndStatusMethods &
+    ({
+      status: (
+        statusCode: 200 | 201
+      ) => NextApiResponseWithoutJsonAndStatusMethods & {
+        json: Send<
+          RS["jsonResponse"] extends z.ZodTypeAny
+            ? z.infer<RS["jsonResponse"]>
+            : any
+        >
+      }
+      json: Send<
+        RS["jsonResponse"] extends z.ZodTypeAny
+          ? z.infer<RS["jsonResponse"]>
+          : any
+      >
+    } & {
+      status: (
+        statusCode: number
+      ) => NextApiResponseWithoutJsonAndStatusMethods & {
+        json: Send<any>
+      }
+      json: Send<any>
+    })
```

Comment:
we should introduce some type tests for this, also if there's a way to break this out for a bit more readability e.g. `SuccessfulApiResponse` and `ErrorApiResponse`

Reply:
yep that's fine

Reply:
note that there are existing tests https://github.com/seamapi/nextlove/blob/main/packages/nextlove/tests/route-spec-types.ts